models/icepack/model_mod.f90

```diff
@@ -0,0 +1,885 @@
+integer,           intent(out) :: kind_list(:)   ! kind number
+logical, optional, intent(out) :: update_var(:) ! logical update
+
+integer :: nrows, i
+character(len=NF90_MAX_NAME) :: varname, dartstr, update
+
+if ( .not. module_initialized ) call static_init_model
+
+nrows = size(table,1)
+
+ngood = 0
+
+! Including this check as well because the conditionals below do not pass and cause the
+! code to error out when the model_state_variables namelist entry is completely empty
+! (model_state_variables = '')
+if ( state_variables(1) == ' ' ) then ! no model_state_variables namelist entry provided
+   string1 = 'model_nml:model_state_variables not specified'
+   call error_handler(E_ERR, 'verify_state_variables', string1, source)
+endif
+
+MyLoop : do i = 1, nrows
+
+   varname = trim(state_variables(3*i -2))
+   dartstr = trim(state_variables(3*i -1))
+   update  = trim(state_variables(3*i   ))
+
+   call to_upper(update)
+
+   table(i,1) = trim(varname)
+   table(i,2) = trim(dartstr)
+   table(i,3) = trim(update)
+
+   if ( table(i,1) == ' ' .and. table(i,2) == ' ' .and. table(i,3) == ' ') exit MyLoop
+
+   if ( table(i,1) == ' ' .or. table(i,2) == ' ' .or. table(i,3) == ' ' ) then
+      string1 = 'model_nml:model_state_variables not fully specified'
+      call error_handler(E_ERR, 'verify_state_variables', string1, source)
+   endif
+
+   ! Make sure DART kind is valid
+   kind_list(i) = get_index_for_quantity(dartstr)
+   if( kind_list(i)  < 0 ) then
+      write(string1,'(''there is no obs_kind <'',a,''> in obs_kind_mod.f90'')') trim(dartstr)
+      call error_handler(E_ERR, 'verify_state_variables', string1, source)
+   endif
+
+   ! Make sure the update variable has a valid name
+   if ( present(update_var) )then
+      SELECT CASE (update)
+         CASE ('UPDATE')
+            update_var(i) = .true.
+         CASE ('NO_COPY_BACK')
+            update_var(i) = .false.
+         CASE DEFAULT
+            write(string1,'(A)')  'only UPDATE or NO_COPY_BACK supported in model_state_variable namelist'
+            write(string2,'(6A)') 'you provided : ', trim(varname), ', ', trim(dartstr), ', ', trim(update)
+            call error_handler(E_ERR, 'verify_state_variables', string1, source, text2=string2)
+      END SELECT
+   endif
+
+   ! Record the contents of the DART state vector
+   if (do_output()) then
+      write(string1,'(A,I2,6A)') 'variable ',i,' is ',trim(varname), ', ', trim(dartstr), ', ', trim(update)
+      call error_handler(E_MSG, 'verify_state_variables', string1, source)
+   endif
+
+   ngood = ngood + 1
+
+enddo MyLoop
+
+end subroutine verify_state_variables
+
+!------------------------------------------------------------------
+! Given a DART location (referred to as "base") and a set of
+! candidate locations & kinds (locs, loc_qtys/indx), returns the
+! subset close to the base, their indices, and their distances to
+! the base
+
+subroutine get_close_state(filt_gc, base_loc, base_type, locs, loc_qtys, loc_indx, &
```

Comment:
Is this doing anything other than the location_mod get_close_state?
At first glance it does not look like it does any special distance calculations. 
You can pass though the location_mod get_close_state